osu.Game.Modes.Osu/Objects/Drawables/DrawableSlider.cs

```diff
@@ -107,7 +107,7 @@ protected override void Update()
 
             if (repeat > currentRepeat)
             {
-                if (ball.Tracking)
+                if (repeat < slider.RepeatCount && ball.Tracking)

```

Comment:
if i recall correctly, the hitobject-level soundtype is to be played on completion, so i'd rather it used the underlying logic instead of "repeat" hitsound logic).